snowflake/src/main/java/org/apache/iceberg/snowflake/JdbcSnowflakeClient.java

```diff
@@ -125,6 +130,15 @@ public <T> T query(Connection conn, String sql, ResultSetParser<T> parser, Strin
   private final JdbcClientPool connectionPool;
   private QueryHarness queryHarness;
 
+  @VisibleForTesting
```

Comment:
I would move these static declarations to the top of the class.  The ordering here is a little awkward.